dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { useState, useRef, useEffect } from 'react';
+
+import {
+  EuiBasicTable,
+  EuiSpacer,
+  EuiLink,
+} from '@elastic/eui';
+
+
+interface TableData {
+  savedHistory: any;
+  savedQuerySearch: (searchQuery: string, selectedDateRange: [], selectedTimeStamp, selectedFields: []) => void;
+}
+
+export function Table(options: TableData) {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const hisRef = useRef();
+  hisRef.current = pageIndex;
+  const thisRef = useRef();
+  thisRef.current = pageSize;
+
+
+  const onTableChange = ({ page = {} }) => {
+    const { index: pageIndex, size: pageSize } = page;
+
+    setPageIndex(pageIndex);
+    setPageSize(pageSize);
+  };
+
+  const columns = [
+    {
+      field: 'data',
+      name: 'Name',
+      render: (item)=>{return <EuiLink onClick={() =>
+      {options.savedQuerySearch(item.query, [item.date_start, item.date_end], item.timestamp, item.fields)}}>
+        {item.name}
+      </EuiLink>},
+    },
+    {
+      field: 'description',
+      name: 'Description',
+    },
+  ];
+
+
+
+  const queries = options.savedHistory.map((h) => {
+    return {
+      data: {
+        name: h?.savedVisualization?.name || h?.savedQuery?.name || '',
+        query: h?.savedVisualization?.query || h?.savedQuery?.query || '',
+        date_start: h?.savedVisualization?.selected_date_range?.start || h?.savedQuery?.selected_date_range.start || '',
+        date_end : h?.savedVisualization?.selected_date_range?.end || h?.savedQuery?.selected_date_range.end ||'',
+        timestamp: h?.savedVisualization?.selected_timestamp?.name || h?.savedQuery?.selected_timestamp?.name || '',
+        fields: h?.savedQuery?.selected_fields?.tokens || h?.savedVisualization?.selected_fields?.tokens || []
```

Comment:
A Better way to do this is to check, does the object 'h' have the field savedVisualization or field savedQuery. You may check this with
```
const savedObject = h.hasOwnProperty('savedVisualization')?  h.savedVisualization : h.savedOuery;
```